server/src/pages/shot/editor.js

```diff
@@ -61,10 +176,246 @@ exports.Editor = class Editor extends React.Component {
+    croppedContext.drawImage(this.imageCanvas, x1, y1, croppedImage.width, croppedImage.height, 0, 0, croppedImage.width, croppedImage.height);
+    croppedContext.globalCompositeOperation = 'multiply';
+    croppedContext.drawImage(this.highlighter, x1, y1, croppedImage.width, croppedImage.height, 0, 0, croppedImage.width, croppedImage.height);
+    let img = new Image();
+    let imageContext = this.imageCanvas.getContext('2d');
+    img.crossOrigin = 'Anonymous';
+    let width = cropWidth;
+    let height = cropHeight;
+    img.onload = () => {
+      imageContext.drawImage(img, 0, 0, width, height);
+    }
+    this.imageContext = imageContext;
+    img.src = croppedImage.toDataURL("image/png");
+    this.canvasWidth = cropWidth;
+    this.canvasHeight = cropHeight;
+    this.onClickCancelCrop();
+  }
+
+  onClickCancelCrop() {
+    this.removeCropBox();
+    this.cropToolBar = null;
+    this.setState({tool: 'pen'});
+  }
+
+  mouseup(e) {
+    if (!selectedPos.width || !selectedPos.height) {
+      this.removeCropBox();
+    }
+    mousedown = false;
+    creating = false;
+    resizing = false;
+  }
+
+  mousedown(e) {
+    mousedown = true;
+    let rect = this.cropContainer.getBoundingClientRect();
+    if (!this.cropBox) {
+      creating = true;
+      mousedownPos.top = e.clientY - rect.top;
+      mousedownPos.left = e.clientX - rect.left;
+    } else {
+      selectedPos.sortCoords();
+      let direction = this.findClickedArea(e);
+      if (direction) {
+        resizing = true;
+        creating = false;
+        resizeDirection = direction;
+        resizeStartPos = {y: e.clientY - rect.top, x: e.clientX - rect.left}
+        resizeStartSelected = selectedPos.clone()
+        this.resizeCropBox(e);
+      }
+    }
+  }
+
+  mousemove(e) {
+    let rect = this.cropContainer.getBoundingClientRect();
+    if (mousedown && creating) {
+      selectedPos = new Selection(
+        mousedownPos.left,
+        mousedownPos.top,
+        Math.min(this.canvasWidth, e.clientX - rect.left),
+        Math.min(this.canvasHeight, e.clientY - rect.top)
+      )
+      selectedPos.top = this.truncateY(selectedPos.top);
+      selectedPos.left = this.truncateY(selectedPos.left);
+      if (selectedPos.width > minWidth && selectedPos.height > minHeight) {
+        this.displayCropBox(selectedPos);
+      }
+    }
+    if (mousedown && resizing) {
+      this.resizeCropBox(e);
+    }
+  }
+
+  resizeCropBox(event, direction) {
+    let rect = this.cropContainer.getBoundingClientRect();
+    let diffX = event.clientX - rect.left - resizeStartPos.x;
+    let diffY = event.clientY - rect.top - resizeStartPos.y;
+    let movement = movementPositions[resizeDirection];
+    if (movement[0]) {
+        let moveX = movement[0];
+        moveX = moveX == "*" ? ["x1", "x2"] : [moveX];
+        for (let moveDir of moveX) {
+          selectedPos[moveDir] = this.truncateX(resizeStartSelected[moveDir] + diffX);
```

Comment:
This should use the `left` and `right` setters in `selectedPos`.

Reply:
This builds on the code for the Screenshots selection box - the Coords are sorted based on the values of `x1`, `x2` and `y1`, `y2`. Using getters and setters breaks resizing.